web/src/main/java/com/bazaarvoice/emodb/web/auth/ApiKeyAdminTask.java

```diff
@@ -332,7 +352,8 @@ private void migrateApiKey(ImmutableMultimap<String, String> parameters, PrintWr
         output.println("\nWarning:  This is your only chance to see this key.  Save it somewhere now.");
     }
 
-    private void deleteApiKey(ImmutableMultimap<String, String> parameters, PrintWriter output) {
+    private void deleteApiKey(Subject subject, ImmutableMultimap<String, String> parameters, PrintWriter output) {
+        subject.checkPermission(Permissions.deleteApiKey());
         String key = getValueFromParams("key", parameters);
```

Comment:
I know that the user will not provide the roles when deleting the api-key, so no roles in the parameters input, but do you think we need to pull up all the roles of the api-key and do a  subject.checkPermission(Permissions.grantRole(RoleIdentifier.fromString(role)));
for each role before deleting the api-key:?